panel/pane/deckgl.py

```diff
@@ -162,7 +162,8 @@ def _update_sources(cls, json_data, sources: list[ColumnDataSource]):
         # Create index of sources by columns
         source_columns = defaultdict(list)
         for i, source in enumerate(sources):
-            key = tuple(sorted(source.data.keys()))
+            # source.data = cast("DataDict", source.data)
```

Comment:
This didn't work, failed with the error: `ValueError: 
ColumnDataSource.data properties may only be set from plain Python dicts,
not other ColumnDataSource.data values.`

Maybe `cast` could be supported in Bokeh?

Reply:
An idea would be to make `source.data = source.data` (which is what cast is) a no-op by doing an identity check. I haven't looked at the Bokeh code to see if this is feasible.